tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/TestSourcesTest.java

```diff
@@ -126,4 +133,72 @@ public Object[][] getExprPos() {
+                .filter(docId -> docId.toString().contains("baz.bal"))
+                .findFirst().get();
+
+        SyntaxTree bazSyntaxTree = baz.document(bazConstantsDocId).syntaxTree();
+
+        // `symbol(Node node)` and `typeOf(Node node)`
+        // use `fooSemanticModel` for `baz` module
+        bazSyntaxTree.rootNode().accept(getNodeVisitor(fooSemanticModel));
+
+        Document bazConstantsDoc = baz.document(bazConstantsDocId);
+
+        // `symbol(Document sourceDocument, LinePosition position)`
+        Optional<Symbol> personRecSymbol = fooSemanticModel.symbol(bazConstantsDoc, from(16, 13));
+        assertTrue(personRecSymbol.isEmpty());
+
+        // `visibleSymbols(Document sourceFile, LinePosition position, DiagnosticState... states)`
+        List<Symbol> visibleSymbols = fooSemanticModel.visibleSymbols(bazConstantsDoc, from(16, 23));
+        assertTrue(visibleSymbols.isEmpty());
+
+        // `references(Document sourceDocument, LinePosition position)`
+        List<Location> references1 = fooSemanticModel.references(bazConstantsDoc, from(0, 0));
+        assertTrue(references1.isEmpty());
+
+        // `references(Document sourceDocument, LinePosition position, boolean withDefinition)`
+        List<Location> references2 = fooSemanticModel.references(bazConstantsDoc, from(0, 0), true);
+        assertTrue(references2.isEmpty());
+
+        // `references(Symbol symbol, Document targetDocument, boolean withDefinition)`
+        Optional<Symbol> constSym = bazSemanticModel.symbol(bazConstantsDoc, from(16, 13));
+        assertTrue(constSym.isPresent());
+        List<Location> references3 = fooSemanticModel.references(constSym.get(), bazConstantsDoc, true);
+        assertTrue(references3.isEmpty());
+
+        // `references(Document sourceDocument, Document targetDocument, LinePosition position, boolean withDefinition)`
+        List<Location> references4 = fooSemanticModel.references(bazConstantsDoc, bazConstantsDoc, from(16, 13), true);
+        assertTrue(references4.isEmpty());
+
+        // `expectedType(Document sourceDocument, LinePosition linePosition)`
+        Optional<TypeSymbol> expType = fooSemanticModel.expectedType(bazConstantsDoc, from(16, 18));
+        assertTrue(expType.isEmpty());
+    }
+
+    private NodeVisitor getNodeVisitor(SemanticModel semanticModel) {
+        return new NodeVisitor() {
+            @Override
+            public void visit(ConstantDeclarationNode constantDeclarationNode) {
+                Optional<Symbol> foundSymbol = semanticModel.symbol(constantDeclarationNode);
+                assertTrue(foundSymbol.isEmpty());
+                constantDeclarationNode.initializer().accept(this);
+            }
+
+            @Override
+            public void visit(BasicLiteralNode basicLiteralNode) {
+                Optional<TypeSymbol> foundTypeSymbol = semanticModel.typeOf(basicLiteralNode);
+                assertTrue(foundTypeSymbol.isEmpty());
```

Comment:
Can we merge these two lines?
```suggestion
                assertTrue(semanticModel.typeOf(basicLiteralNode).isEmpty());
```